### DISCUSSION TITLE
v1.21.0 release and project rename to Vaultwarden

### DISCUSSION THREAD

**Discussion by dani-garcia** [dani-garcia/vaultwarden#1642] [General]
# Project renamed to Vaultwarden
Due to user confusion and to avoid any possible trademark/brand issues with the official server, this project is going to be renamed to Vaultwarden. This rename might mean you will need to make some changes to your setup.

## Docker users
- If you are using any of the docker images, you will have to migrate to the new image under the Vaultwarden organization: 
        
    ####    `vaultwarden/server`

The new image is equivalent to the old one except for the name, so the upgrade is as simple as removing the old container and creating a new one with the new image, making sure the rest of parameters are the same. For example, if you used the default `docker run` command mentioned in the readme, and upgrade would be done like this:
```sh
# Stop and remove the old container
docker stop bitwarden
docker rm bitwarden

# Start new container with the data mounted, make sure the path is the same as the previous one
docker run -d --name vaultwarden -v /bw-data/:/data/ -p 80:80 vaultwarden/server:latest
```
Note that the old images aren't gone yet, just deprecated. They will keep being updated for another two releases after this one to avoid breakage, but it's recommended to migrate to the new image as soon as possible.

If you are still using the old database specific images (server-mysql and server-postgresql), they are also deprecated since the main server image supports all three databases at the same time, [more info on release 1.17.0](https://github.com/dani-garcia/vaultwarden/releases/tag/1.17.0).

If you are using the startup scripts mounted inside the container (`/etc/bitwarden_rs.sh` and `/etc/bitwarden_rs.d/*`), they will now load with the Vaultwarden name first and the bitwarden_rs name as a fallback for compatibility, but it's recommended to rename them as well.

## Git users
If you are using the git repository, we've used this opportunity to rename the `master` branch to `main`, so make sure to have that in mind. The rest of the process is the same and GitHub will redirect the old remote URL to the new one, simply note that the binary generated by cargo is now named `vaultwarden` instead of `bitwarden_rs`.

# Changelog
- Add support for enabling auto-deletion of trash items after X days, disabled by default
    - Set `TRASH_AUTO_DELETE_DAYS` to a positive value to enable this functionality
    - You can also configure how often this process runs, using cron sintax with the variable `TRASH_PURGE_SCHEDULE`
- Updates to the icon fetching, making it more reliable in detecting icon types
- Updated admin page, improving version checks and SQLite backup feature

<hr /><em>This discussion was created from the release <a href='https://github.com/dani-garcia/vaultwarden/releases/tag/1.21.0'>1.21.0</a>.</em>

**Comment by oldLucke** [comment]
I'm having trouble using the new version. Even when doing `docker run vaultwarden/server:lastest` I get this: ![image](https://user-images.githubusercontent.com/30210892/116635662-a62ebc00-a95f-11eb-95c7-c95d645a4b90.png) I made sure to prune all old images and pull the new one. I know that the `Starting Bitwarden_RS` was changed so I don't know why this is showing.

**Comment by Qeynos** [comment]
wrong version in admin panal

Server Installed Update Branched
1.20.0-ced7f177 (heads/1.21.0)
Server Latest
1.21.0

**Reply by BlackDex** [reply]
Not sure what happens there, but it's looks like you are using a commit right before the name change. All seems fine. On the VaultWarden docker hub

**Reply by Qeynos** [reply]
vaultwarden/server             latest               87fb6f9f1382   10 hours ago    179MB

docker image 10hours ago

and i check the 1.20 image shows:
Server Installed
1.20.0

**Reply by Simon-official** [reply]
Can confirm, I've pulled `vaultwarden/server:latest` and the admin panel shows the version `1.20.0-ced7f177 (heads/1.21.0)`.

**Reply by mikkelnl** [reply]
Same here. 

Server Installed: 1.20.0-ced7f177 (heads/1.21.0)
Server Latest: 1.21.0

**Reply by MilesTEG1** [reply]
Hello,
I had the same things :
![image](https://user-images.githubusercontent.com/29298312/116665587-3f7ac400-a99a-11eb-9fd4-d8c71bde20a8.png)
With this image : 
![image](https://user-images.githubusercontent.com/29298312/116665658-528d9400-a99a-11eb-8781-84c818d300b1.png)

But this was with the image I pulled yesterday after knowing the changed name.
I recreate the container few minutes ago with pulling new image : 
![image](https://user-images.githubusercontent.com/29298312/116666264-0bec6980-a99b-11eb-8753-470db4380da1.png)

ANd now I've got this : 
![image](https://user-images.githubusercontent.com/29298312/116666214-fecf7a80-a99a-11eb-87a1-42bf8295a962.png)

**Reply by Qeynos** [reply]
fixed in new image 2 hours ago

**Reply by MilesTEG1** [reply]
> fixed in new image 2 hours ago

Haaa ! That explains :)

**Reply by BlackDex** [reply]
I think the alpine image wasn't updated yet. Sometimes docker hub takes a long time to build the images.

**Comment by avoidik** [comment]
All good, thank you. If someone wants to install Vaultwarden on Raspberry Pi I have prepared a very well explained guide on how to do that. https://gist.github.com/avoidik/9f12ef4feae6ccf7a5801a520931c5d1

**Reply by Pofilo** [reply]
Hello @avoidik !
I have a raspberry pi (but I use the Docker container), do you want me to test the procedure ?
You could also add a page in the wiki no ? (I don't know if it is possible, but it could be helpful and people will find the guide easier than here).

**Comment by luckydonald** [comment]
@dani-garcia 

Maybe a properly communication of this would help. I would recommend a section in the readme explaining that
a) Yes, this was the formally name `bitwarden_rs` (which many users trust), just under a different name (which they don't trust yet).
b) How to update
c) Explanation why

Basically I would recommend to place the explanation you did here in a easily visible space in the `README.md`.

---

This forum is really not easy to reach. I finally found it as I tried to open up a new issue which would have had a properly titled version of  "What the heck is going on, this project had a different name just days ago, _aaaaaa_ what do I do?".

**Reply by BlackDex** [reply]
Well the discussions link is just two steps away from the issues link, so it isn't that hard.
But i have created a new Pinned issue, which points to this article.

**Reply by luckydonald** [reply]
For one who never used that feature I wouldn't even have looked at it. But I still think this would be best to include in the README.

**Reply by BlackDex** [reply]
I Also added a pinned topic/banner to the discourse forum.

Maybe adding something to the README is a good idea also.

**Comment by dani-garcia** [comment]
For those getting a wrong version when running the latest image, the images were accidentally built with a slightly older revision of the projects code, right before it was renamed. I've restarted the builds and in an hour or two it should be fixed.

Note that the only differences between the wrong build and the new one are the version number and name, otherwise they are identical.

I also added a small note at the top of the readme mentioning the rename and pointing here, like @luckydonald mentioned.

**Reply by Joulinar** [reply]
Hi, could the version tag incorrect for the Git version (non-Docker) as well? I did a fresh install using `https://github.com/dani-garcia/vaultwarden/archive/1.21.0.tar.gz` but on the web page it is stating `Version 2.19.0` still.

**Reply by BlackDex** [reply]
The web-vault and server-api are two different products both versioned separately.

**Reply by Joulinar** [reply]
thx for clarification

**Comment by UnlimitedCookies** [comment]
I guess the name change differentiates this project further from the official, which is a step in the right direction. (Even though the name vaultwarden is not as catchy as bitwarden :D )

On this occasion you should also rename bw_web_builds (maybe vaultwarden_web?) and update all the descriptions.

@dani-garcia Also don't forget to change the description on your sponsor page.

**Reply by luckydonald** [reply]
And the forum.

**Reply by dani-garcia** [reply]
Just changed the sponsors and web vault builds descriptions, though I'll keep the repo name the same as to me it refers to the bitwarden web client builds for vaultwarden.

Changed the forum descriptions and branding as well, I'll have to contact discourse to check if we can change the subdomain without breaking old links.

**Reply by UnlimitedCookies** [reply]
@BlackDex #1180 Also needs to be updated with the new dockerhub handles.

**Comment by remiflament** [comment]
I very much appreciate your work! Thank you for this new release.

**Comment by dhruv130797** [comment]
How do I update Bitwarden to Vaultwarden using portainer?? Can someone please guide me. I don't want to loose any data

**Reply by halloamt** [reply]
Hi,
just get the new image and redeploy the container with the same parameters. Worked like a charm for me.
And you will not loose any data if you have a backup.

**Reply by dhruv130797** [reply]
Ok. Thanks!

**Comment by bvklaveren** [comment]
I happen to have a shared hosting account on Dreamhost, so I found the instructions on https://github.com/jjlin/vaultwarden-shared-hosting extremely helpful. I have installed Bitwarden_rs some time ago and recently upgraded to Vaultwarden.

When I go the the page where I host my Bitwarden/Vaultwarden, I see this at the bottom of the page:
```
© 2021, Bitwarden Inc. (Powered by Vaultwarden)
Version 2.19.0 
```

When I go to the admin > diagnostics page, I see this:
```
Server Installed **Update**: 1.18.0
Server Latest: 1.21.0
Web Installed **Update**: 2.19.0
Web Latest: 2.20.1
```

To me this sounds like I am running an outdated version. I have followed the instructions from https://github.com/jjlin/vaultwarden-shared-hosting#upgrade-vaultwarden and did not get any error messages, but still the version numbers are what they are. So am I doing anything wrong? Should I not see version number 2.20.1?

**Reply by jjlin** [reply]
Vaultwarden is not quite ready for version 2.20.1 of the web vault, and it has not been built into any of the Docker images. The web vault version bundled with the `vaultwarden` executable in a given Docker image is always the recommended version.

**Reply by bvklaveren** [reply]
So what you are saying is that it is normal that I am on 2.19.0 and there is nothing to worry about? I cannot find any current version numbers in Vaultwarden or the docker images.

**Reply by BlackDex** [reply]
Well the web version is ok, but the server version is behind, that is currently on v1.21.0 not v1.18.0. so you probably need to correctly update it.

**Reply by bvklaveren** [reply]
As I could not get the server updated, I decided to export my passwords and delete the sub domain and vaultwarden folders and start over. No matter what I try, it won't work anymore. I tried deleting the subdomain and user, which should delete all associated files. I then created a new and different subdomain and a new user and still it does not work. The only thing I see is this:
`404: Not Found`
`The requested resource could not be found.`
`Rocket`
I am at a total loss. Any ideas?

**Comment by PeterNerlich** [comment]
I'm completely new to bitwarden and this project and am just looking into it. What I am confused about and would wish to be explained in the readme somewhere (but what also could just be associated with the recency of the rename) is what exactly this project is and in how far it is a *replacement* for the bitwarden services or just a lightweight server thing that still has ties to them.
Things I am confused about as a list:

- Branding: Web interface of my spun up container greets me with big "**bit**warden" banner, indistinguishable from the official service. The bottom still has "© 2021, Bitwarden Inc." with the only addition of "(Powered by Vaultwarden)" and the Terms of service and privacy policy link directly to bitwarden.com
- Degree of independence: Is this a completely independent project with the same API or do I still give some of my information to bitwarden?

Another thing I might add is that I had problems finding the hints to how to configure aspects beyond just pulling the image and bringing up a container as is.

**Reply by BlackDex** [reply]
Have you checked the wiki?
https://github.com/dani-garcia/vaultwarden/wiki
https://github.com/dani-garcia/vaultwarden/wiki/FAQs
https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview

**Reply by PeterNerlich** [reply]
Thank you, I kinda did, but was rather impatient with reading because I just wanted to quickly try around a bit in a container.

I think more clarification in the README might be helpful *("This project is not associated [...] and usage does not entail communications to the official service" or something along those lines)* and obviously the branding might be nice, though by now I gathered the web interface is probably just using the upstream component as is, so I understand how this means a bunch of extra work that does not really contribute functionally.
The `.env.template` could be hinted at/linked in the README as well, but I am undecided whether I wouldn't find a wiki page more helpful instead.

**Comment by christiaangoossens** [comment]
According to the Bitwarden trademark guidelines (https://github.com/bitwarden/server/blob/master/TRADEMARK_GUIDELINES.md), one can only use the name Bitwarden, including the logo, anywhere if you are operating the software under subscription or license. Will this change also remove the logo of Bitwarden on the login page? Right now, I am unable to use vaultwarden as a service-provider due to the Bitwarden logo being there..

Alternatively, does anyone have documentation proving otherwise? Very interested to see if there are possibilities to supply vaultwarden as a company service to non-profit organizations at low cost.

**Reply by christiaangoossens** [reply]
_I am aware that you can host your on web vault interface, but doing that with the Vaultwarden patches is a lot of work, and it would likely save everyone trademark issues if we just replace the logo fully upstream in vaultwarden, or alternatively, just remove it entirely._

Additionally, this change concerns three occurances of the logo: the login page front logo, the favicon and the small bitwarden logo on the top blue bar on all pages.

**Reply by BlackDex** [reply]
If i'm not mistaken, @dani-garcia spoken with the founder of Bitwarden regarding this, and they agreed upon adding a `(Powered by Vaultwarden)` message to be enough. Also the web-vault is of course still code from 8Bit, with just minor adjustments, i don't see a big issue in leaving it there. We also are not going to provide custom mobile clients, so having the web-interface keeping the same style/layout seems like the best thing to do i think.

**Reply by dani-garcia** [reply]
Yeah, what I got from Kyle when we talked about the rename was that the footer message in the web vault would be enough.
Initially I left the original branding to keep all the clients with the same branding thinking it would avoid confusion, but I've noticed some people have gotten confused seeing the bitwarden brand in the included web vault, so I'm not so sure about it now.

**Reply by BlackDex** [reply]
Maybe something to change with the new vault version then?
We do need to keep the © notice right? But the rest could be changed i think.

**Reply by christiaangoossens** [reply]
@dani-garcia Could we get some further confirmation from Bitwarden that their trademark guide is wrong then? Because I cannot match up what you heard from Kyle ('its okay') and the trademark guide saying very clearly that we should have a license to operate the Bitwarden trademark/logo's. Thank you for your response btw :)

**Comment by pepa65** [comment]
With all clients using Bitwarden, it would be confusing for users if the vaultwarden web vault had different logos. I would like it if the /admin page had the vaultwarden logos, they look cool. But since we are allowed to keep using the Bitwarden trademarks, name and logos, we should keep using that privilege in order to present a united face to users.

**Reply by BlackDex** [reply]
> I would like it if the /admin page had the vaultwarden logos, they look cool.

That is already the case :).

> But since we are allowed to keep using the Bitwarden trademarks, name and logos, we should keep using that privilege in order to present a united face to users.

I'm a bit two sided on this. Using the vaultwarden logo will make it more clear it's a bit different. It could meen more web-vault issues being reported here and we need to redirect that. But if that is all.

Using the bitwarden logo keeps it more in line with all the other apps.

I'm fine both ways, just not sure which one use the best

**Reply by pepa65** [reply]
> > I would like it if the /admin page had the vaultwarden logos, they look cool.
> 
> That is already the case :).

And indeed! The 2.19.0 has "Vaultwarden" with the Bitwarden shield, and 2.20.4 has the Vaultwarden cogwheel..!
Users don't want it to be different, it's only admins that care about that. Users need it to be the same, for trust reasons, especially in something as security sensitive as a password manager.

**Reply by Pofilo** [reply]
> Using the bitwarden logo keeps it more in line with all the other apps.

Totally agree, it would be confusing to have the web page with a logo and all other apps with the Bitwarden Logo.

**Reply by christiaangoossens** [reply]
Agree that it would maybe be confusing, however, if having the trademarked logo on your web vault requires a license according to Bitwarden's trademark guide, it would likely be better to remove all logos alltogether, and just note that it's "Based on Bitwarden" (like the trademark guide suggests).

**Reply by BlackDex** [reply]
Well, they do provide self-hosted instances them self with all the logo's and stuff. And since they agreed upon the addition of `(Powered by Vaultwarden)` would be enough i do not think there is any issue with using those images at all.
If you really feel strong on not having them on your own instance, you can always replace them your self during the startup of your instance by either overwriting them via a `-v` or a script placed in a volume mounted on `/etc/vaultwarden.d`.

**Reply by pepa65** [reply]
> you can always replace them your self during the startup of your instance by either overwriting them via a `-v` or a script placed in a volume mounted on `/etc/vaultwarden.d`.

I maintain such a solution here: https://gitlab.com/pepa65/brandvaultwarden (It doesn't feature the Vaultwarden logo's, you'd have to source those yourself..!)

[a comment by an automated account is not shown]

**Comment by sfkpmr** [comment]
Is it intentional that exports are named bitwarden_export_[date].json?

**Reply by BlackDex** [reply]
That is how the web-vault exports are generated. Also, it is a bitwarden format, so i think that would be correct.

**Comment by MrLatroX** [comment]
Hey,
did anyone know how i can do this in Plesk?
I use Docker in my Plesk Installation and there i can't change the source.

**Reply by avoidik** [reply]
Just mirror docker container image where Plesk is able to get it

**Comment by pepa65** [comment]
@MrLatroX We do this in Plesk under docker: https://gitlab.com/pepa65/brandvaultwarden

[a comment by an automated account is not shown]

**Comment by estranged-networks** [comment]
@dani-garcia , not to be the one who starts adding more crap here @Ornias1993 , but ....I've upgraded my bitwardenrs to vaultwarden eventually and , Boom ! no data , cant log in , doesn't recognise any of my user accounts :( - I have a backup & reverting that as I type this up - I was on 2.19.0 running in docker on a rasp pi  -  any ideas what could have gone wrong ... it was just a stop, rm, create new container :( . Any advise would be greatly appreciated

**Reply by BlackDex** [reply]
@estranged-networks i suggest to start a new discussion with as much as details as you can give like logs, what do you see if you go to `/admin` for example and not clutter this topic.
